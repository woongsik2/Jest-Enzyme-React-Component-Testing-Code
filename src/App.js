import React, { Component } from 'react';
import Counter from './components/Counter.jsx';
import NameForm from './components/NameForm.jsx';
import NameList from './components/NameList.jsx';

class App extends Component {
  state = {
    names: ['웅식', '킴웅식']
  }
  onInsert = (name) => {
    this.setState(({names}) => ({names: names.concat(name)}));
  }
  render() {
    const {names} = this.state;
    const {onInsert} = this;
    return (
      <div>
        <Counter />
        <hr/>
        <h1>이름 목록</h1>
        <NameForm onInsert={onInsert}/>
        <NameList names={names}/>
      </div>
    );
  }
}

export default App;
