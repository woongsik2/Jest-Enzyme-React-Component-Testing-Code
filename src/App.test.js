import React from 'react';
// import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('App View', () => {
  let component = null;

  it('App View 초기 렌더링', () =>{
    component = renderer.create(<App />);
  });

  it('App View Snapshot check', () =>{
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });

});