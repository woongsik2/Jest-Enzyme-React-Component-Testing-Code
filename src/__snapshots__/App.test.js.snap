// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App View App View Snapshot check 1`] = `
<div>
  <div>
    <h1>
      Counter
    </h1>
    <h2>
      1
    </h2>
    <button
      id="plus"
      onClick={[Function]}
    >
       + 
    </button>
    <button
      id="minus"
      onClick={[Function]}
    >
       - 
    </button>
  </div>
  <hr />
  <h1>
    이름 목록
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      이름
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
  <ul>
    <li>
      웅식
    </li>
    <li>
      킴웅식
    </li>
  </ul>
</div>
`;
